File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/dataprovider/BarLineScatterCandleBubbleDataProvider.java
package com.github.mikephil.charting.interfaces.dataprovider;

import com.github.mikephil.charting.components.YAxis.AxisDependency;
import com.github.mikephil.charting.data.BarLineScatterCandleBubbleData;
import com.github.mikephil.charting.utils.Transformer;

public interface BarLineScatterCandleBubbleDataProvider extends ChartInterface {

    Transformer getTransformer(AxisDependency axis);
    boolean isInverted(AxisDependency axis);
    
    float getLowestVisibleX();
    float getHighestVisibleX();

    float getLowestVisibleY(AxisDependency axis);
    float getHighestVisibleY(AxisDependency axis);

    BarLineScatterCandleBubbleData getData();
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/dataprovider/BarLineScatterCandleBubbleDataProvider.kt
package com.github.mikephil.charting.interfaces.dataprovider

import com.github.mikephil.charting.components.YAxis.AxisDependency
import com.github.mikephil.charting.data.BarLineScatterCandleBubbleData
import com.github.mikephil.charting.utils.Transformer

interface BarLineScatterCandleBubbleDataProvider : ChartInterface {
    fun getTransformer(axis: AxisDependency?): Transformer?
    fun isInverted(axis: AxisDependency?): Boolean
    val lowestVisibleX: Float
    val highestVisibleX: Float
    override fun getData(): BarLineScatterCandleBubbleData<*>
}